### PR TITLE
Summarize published data

### LINKED ISSUE #540: [FEATURE] Create taxonomic view page
This page which show all object categories people have annotated in the system (or others manually entered) and let people organize them into a hierarchy of labels, defining child/parent relationships or label synonyms

### LINKED ISSUE #648: [FEATURE] Add side page that lists all object categories in all public folders
As a follow-up, the ability to click the object label and see any videos with annotations on them in a list would be helpful like

video name - video user  with links

### PR DESCRIPTION
**Pull request by subdavis**
Screenshot from 2021-03-26 13-53-41

Adds new dependency `color-hash` to generate consistent colors by datasetId so you can see at-a-glance where the same dataset appears for multiple types in the list.

fixes #540 
fixes #648

### CODE REVIEW

**Review by BryonLewis — CHANGES_REQUESTED** [review]
Overall it works very well.  Just a few questions/comments.  I think depending on the opt-in/opt-out discussion for users if it becomes opt-in we may want to make it easier for them to tag their public data using a context menu in GWC or something else on the folder.

File: server/dive_server/constants.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "video/x-msvideo",
 }
 
-
+PublishedMarker = 'published'
```
[the remark this answers is not in the document]
Comment (subdavis) [reply server/dive_server/constants.py:38]: Yes, thank you.

File: server/dive_utils/models.py [hunk 2]
```diff
@@ -81,6 +81,17 @@ class Config:
         extra = 'forbid'
 
 
+class SummaryItemSchema(BaseModel):
+    value: str
+    total_tracks: int
+    total_detections: int
+    found_in: List[str]
```
[the remark this answers is not in the document]
Comment (subdavis) [reply server/dive_utils/models.py:88]: There's insufficient horizontal space to use anything but a short segment of the ID IMO.

We could include the name, but there's nowhere on the page to put it.

File: client/platform/web-girder/views/Summary.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            </td>
+            <td>
+              {{ item.total_tracks }}
+            </td>
+            <td>
+              {{ item.total_detections }}
+            </td>
+            <td>
+              <v-btn
+                v-for="datasetId in item.found_in"
```
[the remark this answers is not in the document]
Comment (subdavis) [reply client/platform/web-girder/views/Summary.vue:53]: That was as far as my plan went for now.

[the remark this answers is not in the document]
Comment (BryonLewis) [reply client/platform/web-girder/views/Summary.vue:53]: yeah that is completely fine.  We can deal with it if it becomes an actual problem.

File: client/package.json [hunk 4]
```diff
@@ -39,10 +39,9 @@
     "@mdi/font": "^5.3.45",
     "@sentry/browser": "^5.24.2",
     "@sentry/integrations": "^5.24.2",
-    "@types/mousetrap": "^1.6.3",
-    "@types/source-map": "0.5.2",
     "@vue/composition-api": "^1.0.0-beta.18",
     "axios": "^0.21.1",
+    "color-hash": "^1.0.3",
```
[the remark this answers is not in the document]
Comment (subdavis) [reply client/package.json:44]: As far as I can tell, that's a finite number of colors.  I need MOAR.  Is there a d3 option more similar to color hash?

[the remark this answers is not in the document]
Comment (BryonLewis) [reply client/package.json:44]: yeah I think you have to sort of build it by using a color scale range as the input for the scale ordinal.  You then have an issue where you could possible get similar colors based on how it does the hashing.  color-hash just works so I think it's easier to use that.  There is some upper limit where eventually the colors aren't different enough.

**Review by BryonLewis — APPROVED** [review]

**Pull request merged**